### ISSUE TITLE
Previews of uploaded pictures not showing anymore

### ISSUE THREAD

**Issue by dcardin** [RocketChat/Rocket.Chat#24113]
### Description:

The preview feature of uploaded pictures is broken

### Steps to reproduce:

Upload a picture

### Expected behavior:

The picture is shown as a preview

### Actual behavior:

Retry button, no preview

### Server Setup Information:

- Version of Rocket.Chat Server: 4.3.1
- Operating System: Linux
- Deployment Method:  docker
- Number of Running Instances: 1
- DB Replicaset Oplog: 
- NodeJS Version: v12.22.8
- MongoDB Version: 4.4.11 / wiredTiger (oplog Enabled)

### Client Setup Information

Multiple clients have exhibited the problem

### Relevant logs:

the client sends a request to /file-upload/Ayq8ZXS4PxWei863w/IMG_4408.HEIC (example image)

The server doesn't respond and the server log contains :

heif: Unsupported feature: Unsupported codec (4.3000)

**Comment by mr-karan** [comment]
+1 Facing the same

Using `rocketchat/rocket.chat:4.3.0`.

**Comment by sampaiodiego** [comment]
does it happen to other image types as well? like PNG and JPEG? or just HEIC?

**Comment by mr-karan** [comment]
Yes it's happening to other image types as well. Here's an example of PNG:

![image](https://user-images.githubusercontent.com/5689132/149813053-bb50f293-9a8a-4077-887c-a54c6c9ba985.png)

Network Tab on clicking _Retry_:

![image](https://user-images.githubusercontent.com/5689132/149813281-d26c3027-b7a9-4549-8d38-4b01d74c4863.png)


I noticed that this happens as soon as the S3 Presigned URL gets expired. The default value is 120 seconds, so when I reload the page after 120s, I see this _Retry_ button. I think when reloading the page, it's somehow _not_ generating a new pre-signed URL and using the old one, which leads to this.

**Comment by mr-karan** [comment]
Can this issue get at least an acknowledgement?

**Comment by levycarneiro** [comment]
We experience this issue one and off for at least 1 year now. It affects some people and not others, then who wasn't affected become affected and vice-versa. We weren't able to pin point a scenario that duplicates it.

**Comment by aixtemaadmins** [comment]
we have the problem every couple days, when i restart the httpd the pictures work again

fedora 36 
snap tracking: 5.x/edge 5.0.2

**Comment by f-9t9it** [comment]
This have been a very confusing bug.

Sometimes when admin log-in it prompts us to change Site URL to the FQDN domain name, sometimes it doesn't and image still try to load using localhost:3000

solutions I tried.
1. add our fqdn domain in server hosts file as localhost. No luck.
2. Change siteurl in snap app setting using `sudo snap set rocketchat-server siteurl=http://chat.rocket.com`  and changed nginx config accordingly but no luck. 
3. As sometimes it prompts admin user to change the site url upon login so I digged into it and found `Site URL` under Workscape > Settings > General > Site URL. I changed it back to my FQDN and that seems to resolved and that reminded me of the last time I changed it so that means after everytime snap is restarted the Site URL goes back to `localhost:3000` 


The solution is to use external Site url and internal site url.  This is something I have faced on JellyFin Server too and there they have a good way to solve such situation.
